### PR TITLE
feat: remove deprecation for save for later feature

### LINKED ISSUE #5958: Breaking changes since 1.5: Save for Later (#5720)
add new services into cart component constructors. the following components are impacted:
- CartDetailsComponent
old one:
```
constructor(
    cartService: CartService,
    // tslint:disable-next-line:unified-signatures
    promotionService: PromotionService,
    selectiveCartService: SelectiveCartService,
    authService: AuthService,
    routingService: RoutingService,
    featureConfig: FeatureConfigService
  );

  /**
   * @deprecated Since 1.5
   * Use promotionService instead of the promotion inputs, add selectiveCartService authService routingService and featureConfig for save for later.
   * Remove issue: #5670 #5958
   */
  constructor(cartService: CartService);

  constructor(
    protected cartService: CartService,
    protected promotionService?: PromotionService,
    protected selectiveCartService?: SelectiveCartService,
    private authService?: AuthService,
    private routingService?: RoutingService,
    private featureConfig?: FeatureConfigService
  ) {}

```
new one:
```
constructor(
    protected cartService: CartService,
    protected promotionService: PromotionService,
    protected selectiveCartService: SelectiveCartService,
    private authService: AuthService,
    private routingService: RoutingService,
  ) {}
```
remove isSaveForLaterEnabled() which is for feature flag.
for ngOnInit, remove old code in else block and only remain new code. 
```
ngOnInit() {
...
    if (this.isSaveForLaterEnabled()) {
//new code
}
else{
//old code
}
```

for the template using cxFeatureLevel to contain new and old logic. Please remove the <ng-container *cxFeatureLevel="'!1.5'"> block and only remain the 'new code'. The <ng-container *cxFeatureLevel="'1.5'"> statement should also be removed.

old code:
```
<ng-container *cxFeatureLevel="'!1.5'">
        //old code
      </ng-container>

      <ng-container *cxFeatureLevel="'1.5'">
        //new code
      </ng-container>
```
new code:
```
<ng-container *ngIf="orderPromotions$ | async as orderPromotions">
          <cx-promotions [promotions]="orderPromotions"></cx-promotions>
        </ng-container>

        <cx-cart-item-list
          [items]="entries"
          [cartIsLoading]="!(cartLoaded$ | async)"
          [promotionLocation]="promotionLocation"
          [options]="{
            isSaveForLater: false,
            optionalBtn:  saveForLaterbtn 
          }"
        ></cx-cart-item-list>
```

- CartItemListComponent
for template, remove  <ng-container *cxFeatureLevel="'!1.5'"> block with old code and remain new code, also need to remove the <ng-container *cxFeatureLevel="'1.5'"> statement.
```
<ng-container *cxFeatureLevel="'!1.5'">
       //old code
      </ng-container>

      <ng-container *cxFeatureLevel="'1.5'">
     //new code
      </ng-container>
```

replace old code 
```
<ng-container
      *ngIf="
        isSaveForLaterEnabled() && options.isSaveForLater;
        else totalHeader
      "
    >
      <div class="cx-item-list-total col-md-2 col-lg-2 col-xl-2">
        {{ 'saveForLaterItems.stock' | cxTranslate }}
      </div>
    </ng-container>
```

to new one:
```
<ng-container
      *ngIf="
        options.isSaveForLater;
        else totalHeader
      "
    >
      <div class="cx-item-list-total col-md-2 col-lg-2 col-xl-2">
        {{ 'saveForLaterItems.stock' | cxTranslate }}
      </div>
    </ng-container>
```
Note: isSaveForLaterEnabled() should be removed from the component.

constructor old one
```
constructor(
    cartService: CartService,
    fb: FormBuilder,
    selectiveCartService: SelectiveCartService,
    featureConfig: FeatureConfigService
  );

  /**
   * @deprecated Since 1.5
   * Add selectiveCartService authService routingService and featureConfig for save for later.
   * Remove issue: #5958
   */
  constructor(cartService: CartService, fb: FormBuilder);

  constructor(
    protected cartService: CartService,
    protected fb: FormBuilder,
    protected selectiveCartService?: SelectiveCartService,
    private featureConfig?: FeatureConfigService
  ) {}
```

new
```
constructor(
    protected cartService: CartService,
    protected fb: FormBuilder,
    protected selectiveCartService: SelectiveCartService,
  ) {}
```

and remove isSaveForLaterEnabled().

- CartItemComponent
old constructor
```
constructor(
    promotionService: PromotionService,
    // tslint:disable-next-line:unified-signatures
    featureConfig: FeatureConfigService
  );
  /**
   * @deprecated Since 1.5
   * Add featureConfig for save for later.
   * Remove issue: #5958
   */
  constructor(promotionService: PromotionService);

  constructor(
    protected promotionService: PromotionService,
    private featureConfig?: FeatureConfigService
  ) {}
```
new one should be
```
constructor(
    protected promotionService: PromotionService,
  ) {}
```

Remove isSaveForLaterEnabled() which is for feature flag.

in the template, update old one:
```
<ng-container
        *ngIf="isSaveForLaterEnabled() && options.isSaveForLater; else total"
      >
//old code
</ng-container>
```
to new:
```
<ng-container
        *ngIf="options.isSaveForLater; else total"
      >
//old code
</ng-container>
```

- CartPageLayoutHandler
old constructor:
```
constructor(
    cartService: CartService,
    selectiveCartService: SelectiveCartService,
    featureConfig: FeatureConfigService
  );
  /**
   * @deprecated Since 1.5
   * Add selectiveCartService and featureConfig for save for later.
   * Remove issue: #5958
   */
  constructor(cartService: CartService);

  constructor(
    private cartService: CartService,
    private selectiveCartService?: SelectiveCartService,
    private featureConfig?: FeatureConfigService
  ) {}
```

new constructor:
```
constructor(
    private cartService: CartService,
    private selectiveCartService: SelectiveCartService,
  ) {}
```
in the handle method, old:
```
if (this.featureConfig && this.featureConfig.isEnabled('saveForLater')) {
        return combineLatest(
//new code)
}
return combineLatest(
//old code
)
```
new should remove feature flag and remain the new code, remove old return:
```
  return combineLatest(
//new code)
}
```

### PR DESCRIPTION
**Pull request by psmul**
Closes: #5958

### CODE REVIEW

File: projects/schematics/src/migrations/2_0/constructor-deprecations/data/cart-item-list.component.migration.ts [hunk 1]
```diff
@@ -87,4 +87,46 @@ export const CART_ITEM_LIST_COMPONENT_MIGRATIONS: ConstructorDeprecation[] = [
       },
     ],
   },
+  {
```
Comment (tobi-or-not-tobi) [inline projects/schematics/src/migrations/2_0/constructor-deprecations/data/cart-item-list.component.migration.ts:90]: @znikola would you mind evaluating this file? I'm unsure if the different migration scripts are what we should end up with?

File: projects/schematics/src/migrations/2_0/constructor-deprecations/data/cart-details.component.migration.ts [hunk 2]
```diff
@@ -137,4 +137,82 @@ export const CART_DETAILS_COMPONENT_MIGRATIONS: ConstructorDeprecation[] = [
       },
     ],
   },
+  {
+    class: CART_DETAILS_COMPONENT,
```
Comment (tobi-or-not-tobi) [inline projects/schematics/src/migrations/2_0/constructor-deprecations/data/cart-details.component.migration.ts:141]: @znikola would you mind evaluating this file? I'm unsure if the different migration scripts are what we should end up with?

File: projects/schematics/src/migrations/2_0/constructor-deprecations/data/cart-item.component.migration.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  CART_ITEM_COMPONENT,
+  FEATURE_CONFIG_SERVICE,
+  PROMOTION_SERVICE,
+  SPARTACUS_CORE,
+  SPARTACUS_STOREFRONTLIB,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+// projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.ts
+export const CART_ITEM_COMPONENT_MIGRATIONS: ConstructorDeprecation[] = [
```
Comment (tobi-or-not-tobi) [inline projects/schematics/src/migrations/2_0/constructor-deprecations/data/cart-item.component.migration.ts:11]: @znikola would you mind evaluating this file? I'm unsure if the different migration scripts are what we should end up with?

**Review by tobi-or-not-tobi — DISMISSED** [review]
LGTM, but I've asked @znikola to validate the migration scripts as I wasn't too sure if these are OK. I'm not sure if the various migrations from different angles should be separated or should be merged into a single script.

**Review by znikola — DISMISSED** [review]

**Review by znikola — APPROVED** [review]

**Review by tobi-or-not-tobi — APPROVED** [review]

**Comment by Matejk00** [comment]
Tested through migration doc input, everything works as is described.

**Pull request merged**